tests/libyui/rspec_tmux_tui.rb

```diff
@@ -0,0 +1,76 @@
+require "shellwords"
+
+class TmuxTui
+  class Error < RuntimeError
+  end
+
+  def self.new_session(*args)
+    new(*args)
+  end
+
+  attr_reader :session_name
+
+  def initialize(shell_command, x: 80, y: 24, detach: true, session_name: nil)
+    @shell_command = shell_command
+    @x = x
+    @y = y
+    @detach = detach
+    @session_name = session_name || new_session_name
+
+    system "tmux", "new-session",
+           "-s", @session_name,
+           "-x", @x.to_s,
+           "-y", @y.to_s,
+           *(@detach ? ["-d"] : [] ),
+           "sh", "-c", "#{@shell_command}; sleep 9999"
+  end
+
+  def new_session_name
+    "tmux-tui-#{rand 10000}"
+  end
+
+  # @return [String]
+  def capture_pane(color: false)
+    esc = color ? "-e" : ""
+    # FIXME: failure of the command?
+    `tmux capture-pane -t #{session_name.shellescape} -p #{esc}`
+  end
+
+  def capture_pane_to(filename)
+    txt = capture_pane(color: false)
+    esc = capture_pane(color: true)
+    File.write("#{filename}.out.txt", txt)
+    File.write("#{filename}.out.esc", esc)
```

Comment:
I'd prefer `.raw` suffix so it's clear that it's the original unprocessed screen dump.

Reply:
Unfortunately, as I understand it, there is no such thing as a raw terminal dump, unless we're talking about the linux console devices (`/dev/vcsa7` having colors for `/dev/vcs7`). I feel `.esc` is more specific.